interpreter/interpreter.go

```diff
@@ -15,23 +15,145 @@ var topScope = NewScopeWithValues(map[string]values.Value{
 	"false": values.NewBool(false),
 })
 
+// Package implementation
+type packageObject struct {
+	name        string
+	scope       *Scope
+	private     map[string]struct{}
```

Comment:
Can you make this a `map[string]bool`? Then you do have to do `_,ok:=pkg.private[foo]` you can just do `ok := pkg.private[foo]`, the zero value is returned when the key does not exists, which in this case means `false`.